packages/smooth_app/lib/pages/preferences/user_preferences_search_page.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:diacritic/diacritic.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/preferences/user_preferences.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/pages/preferences/abstract_user_preferences.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_item.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_page.dart';
+
+/// Search page for preferences, with TextField filter.
+class UserPreferencesSearchPage extends StatefulWidget {
+  const UserPreferencesSearchPage({super.key});
+
+  @override
+  State<UserPreferencesSearchPage> createState() =>
+      _UserPreferencesSearchPageState();
+}
+
+class _UserPreferencesSearchPageState extends State<UserPreferencesSearchPage> {
+  final TextEditingController _controller = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    final UserPreferences userPreferences = context.watch<UserPreferences>();
+    final List<Widget> items = _getItems(_controller.text, userPreferences);
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Preferences Search'),
+      ),
+      body: SafeArea(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            TextField(
```

Comment:
We should (if moved out of the dev mode) switch to the SmoothTextFormField

Reply:
> We should (if moved out of the dev mode) switch to the SmoothTextFormField

Of course the UI/UX should be redesigned once the concept is approved and the code moved beyond the dev mode.